### ISSUE TITLE
Lynis report imports Hostname value into IP Address, creating duplicated hosts

### ISSUE THREAD

**Issue by daniejstriata** [infobyte/faraday#268]
## Issue Type
 - Bug Report

## Faraday version
Faraday v3.0

## Steps to reproduce

When I import a host's Lynis report into Faraday there is a new host added but with the Hostname filled into the IP Address field.
Other reports like Nessus are correctly adding the same host in its IP address and the hostname entry is empty.

Effectively I now have 2 hosts in Faraday for the same host. The one where the IP address is the hostname and the other where the IP is the IP and the hostname is empty. 
![image](https://user-images.githubusercontent.com/13314239/43629229-f805c67c-96fc-11e8-887a-ad194a974c9a.png)

Even by adding the Hostname to the host entry with the IP address does not merge the 2 hosts after the imports complete. 

Lynis created host:
![image](https://user-images.githubusercontent.com/13314239/43628880-cb49001e-96fb-11e8-87ac-0c2d0675171e.png)
Nessus created host:
![image](https://user-images.githubusercontent.com/13314239/43629046-5dd644e6-96fc-11e8-84bb-5bb86e409e92.png)

## Expected results
1. Ideally I would prefer to see the hosts vulnerabilities combined rather than have a separate hostname and an IP to deal with. 
2. Why does running a command like host not update the IP Address's hostname field? Or is there another way to update the IP and hostname fields?
 `host <IP Address>` add the IP and hostname to Faraday hosts if not added yet.
 `host <Host name>` add the hostname and IP to Faraday hosts if not added yet.
3. I added the Hostname to the IP host and was hoping that Faraday would merge the updated IP host with the existing Host with a name.

### OS
$ cat /etc/lsb-release
DISTRIB_ID=LinuxMint
DISTRIB_RELEASE=18.3
DISTRIB_CODENAME=sylvia
DISTRIB_DESCRIPTION="Linux Mint 18.3 Sylvia"

**Comment by WinnaZ** [comment]
Hey there,
First! Are you using v3?
We made some changes around those fields in our last release.
I'll take a look at this plugin and beck back to you.
Cheers

**Comment by daniejstriata** [comment]
I am using version 3.

Faraday v3.0

**Comment by montive** [comment]
Hi there.

We are working on this issue in order to give you an answer asap. 
Let us know if you have any question.

Cheers

**Comment by daniejstriata** [comment]
I've found that when I import a Nessus report and it creates a host with IP X that I cannot change the IP for the host already added by the Lynis report to use the same IP. From the issue I logged above the Lynis import incorrectly added the Hostname to the IP address field.

I think that my update is blocked by design but if it merged my Nessus and Lynis vulnerabilities as the host had the same IP, that would've been a good for me.

**Comment by daniejstriata** [comment]
Was there any feedback from this plugin yet?

**Comment by montive** [comment]
Hi there. 

We invite you to update Faraday to its 3.1 version. 
Import the report once more and let us know how it goes.

We will wait for your answer.
Cheers.

**Comment by daniejstriata** [comment]
Will do. My upgrade from 3 to 3.1 broke. I'll have to check on Monday and get back to you.

BTW
Is there any documented upgrade process?

**Comment by montive** [comment]
Yes there is: https://github.com/infobyte/faraday/wiki/Updates

Ok, keep us posted.
Cheers.

**Comment by daniejstriata** [comment]
I've updated to 3.1.1 and when I import a Lynis report now nothing seems to happens. It seems to do some processing but nothing is imported or errors reported. I ran imports for other reports and they import. Seems Lynis importing is broken in the latest version.

**Did nothing:**
2018-10-15 12:30:13,569 - faraday.ReportProcessor - INFO [reports_managers.py:53 -           sendReport() ]  The file is /home/xxx/Desktop/xxx/xxx-db v4 lynis-report .log, Lynis
2018-10-15 12:30:13,711 - faraday.ModelController - INFO [controller.py:422 -         _pluginStart() ]  Plugin Started: Lynis. 
2018-10-15 12:30:13,711 - faraday.ModelController - INFO [controller.py:429 -           _pluginEnd() ]  Plugin Ended: Lynis

**Worked:**
2018-10-15 12:36:57,640 - faraday.ReportProcessor - INFO [reports_managers.py:53 -           sendReport() ]  The file is /home/xxx/Downloads/xxx_g2onl3.nessus, Nessus
2018-10-15 12:36:57,781 - faraday.ModelController - INFO [controller.py:422 -         _pluginStart() ]  Plugin Started: Nessus. 
2018-10-15 12:36:59,116 - faraday.ModelController - INFO [controller.py:429 -           _pluginEnd() ]  Plugin Ended: Nessus

**Comment by daniejstriata** [comment]
Should I log this as a new bug? Lynis is not working on the later versions for me.

**Comment by llazzaro** [comment]
Hi @daniejstriata,

We confirm that the following error is a bug and will be fixed on v3.4 (December):
```
 File "/home/xxx/.faraday/plugins/lynis/plugin.py", line 168, in parseOutputString
    ip = socket.gethostbyname(hostname)
gaierror: [Errno -2] **Name or service not known**
```

The last error is a permission error on the file /var/log/lynis-report.dat, make your user can read that file.

**Comment by llazzaro** [comment]
@daniejstriata We released the fix of the **gaierror: [Errno -2] **Name or service not known**** on v3.4

**Comment by llazzaro** [comment]
This issue was fixed in v3.5
Please reopen if you still have this issue.
thanls!
